flopy/mf6/data/mfdataplist.py

```diff
@@ -0,0 +1,2527 @@
+                        )
+                    data_storage.binary = data_record["binary"]
+                if "data" in data_record:
+                    # data gets written out to file
+                    self._set_data(data_record["data"])
+                    # get file path
+                    fd_file_path = self._get_file_path()
+                    # make sure folder exists
+                    folder_path = os.path.split(fd_file_path)[0]
+                    if not os.path.exists(folder_path):
+                        os.makedirs(folder_path)
+                    # store data
+                    self._write_file_entry(fd_file_path)
+            else:
+                if "data" in data_record:
+                    data_storage.modified = True
+                    data_storage.set_internal(None)
+                    self._set_data(data_record["data"])
+            if "iprn" in data_record:
+                data_storage.iprn = data_record["iprn"]
+
+    def append_data(self, data):
+        """Appends "data" to the end of this list.  Assumes data is in a format
+        that can be appended directly to a pandas dataframe.
+
+        Parameters
+        ----------
+            data : list(tuple)
+                Data to append.
+
+        """
+        try:
+            self._resync()
+            if self._get_storage() is None:
+                self._data_storage = self._new_storage()
+            data_storage = self._get_storage()
+            if (
+                data_storage.data_storage_type
+                == DataStorageType.internal_array
+            ):
+                # update internal data
+                self._set_data(data, append=True)
+            elif (
+                data_storage.data_storage_type == DataStorageType.external_file
+            ):
+                # get external data from file
+                external_data = self._get_dataframe()
+                if isinstance(data, list):
+                    # build dataframe
+                    data = pandas.DataFrame(
+                        data, columns=external_data.columns
+                    )
+                # concatenate
+                data = pandas.concat([external_data, data])
+                # store
+                ext_record = self._get_record()
+                ext_record["data"] = data
+                self._set_record(ext_record)
+        except Exception as ex:
+            type_, value_, traceback_ = sys.exc_info()
+            raise MFDataException(
+                self.structure.get_model(),
+                self.structure.get_package(),
+                self._path,
+                "appending data",
+                self.structure.name,
+                inspect.stack()[0][3],
+                type_,
+                value_,
+                traceback_,
+                None,
+                self._simulation_data.debug,
+                ex,
+            )
+
+    def append_list_as_record(self, record):
```

Comment:
Yes I could potentially do this, though this would add complexity to the code.  Telling Flopy that the list passed in is a single record makes flopy not have to determine whether the data is a single record or multiple records.  This check is not trivial since a single record could potentially have a cellid tuple in it (old format).  I prefer to use the less complex solution in this case.